### PR TITLE
dont integrate higher than weather model

### LINKED ISSUE #574: [BUG] Unusual Nodata in GUNW workflow for HRRR
Using the develop branch of RAiDER and the GUNW workflow via:

```
raider.py ++process calcDelaysGUNW -f S1-GUNW-A-R-064-tops-20210723_20210711-015000-00119W_00033N-PP-6267-v2_0_6.nc -m HRRR
```

We get the following `CRITICAL` warning:

```
...
Processing slice 1 / 20: -500.0
Processing slice 2 / 20: 0.0
Processing slice 3 / 20: 500.0
Processing slice 4 / 20: 1000.0
Processing slice 5 / 20: 1500.0
Processing slice 6 / 20: 2000.0
Processing slice 7 / 20: 2500.0
Processing slice 8 / 20: 3000.0
Processing slice 9 / 20: 3500.0
Processing slice 10 / 20: 4000.0
Processing slice 11 / 20: 4500.0
Processing slice 12 / 20: 5000.0
Processing slice 13 / 20: 5500.0
Processing slice 14 / 20: 6000.0
Processing slice 15 / 20: 6500.0
Processing slice 16 / 20: 7000.0
Processing slice 17 / 20: 7500.0
Processing slice 18 / 20: 8000.0
Processing slice 19 / 20: 8500.0
Processing slice 20 / 20: 9000.0
CRITICAL: There are missing delay values. Check your inputs.    
```

Note the product above can be downloaded [here](https://github.com/dbekaert/RAiDER/blob/azimuth-time-interpolation/test/gunw_azimuth_test_data/S1-GUNW-A-R-064-tops-20210723_20210711-015000-00119W_00033N-PP-6267-v2_0_6.nc) (this is test data on an open PR).

Here is the product [link](https://gunw-development-testing.s3.us-west-2.amazonaws.com/raider-gunw/S1-GUNW-A-R-064-tops-20210723_20210711-015000-00119W_00033N-PP-6267-v2_0_6__NODATA.nc) that I get after running the above.

Here is a plot of a level with nodata:

image

Here is the code I used to plot the above and all height layers: 

```
import xarray as xr
import matplotlib.pyplot as plt
path = 'S1-GUNW-A-R-064-tops-20210723_20210711-015000-00119W_00033N-PP-6267-v2_0_6.nc'
for ifg_type in ['reference', 'secondary']:
    for var in ['troposphereHydrostatic', 'troposphereWet']:
        group = f'science/grids/corrections/external/troposphere/HRRR/{ifg_type}'
        with xr.open_dataset(path, group=group) as ds:
            da = ds[var]
       
        for h in da.heightsMeta:
            da.sel(heightsMeta=h).plot()
            plt.show()
```

Here is the full output:

```
(RAiDER) --- Desktop/issue-ticket-gunw » raider.py ++process calcDelaysGUNW -f S1-GUNW-A-R-064-tops-20210723_20210711-015000-00119W_00033N-PP-6267-v2_0_6.nc -m HRRR
Downloading S1B_OPER_AUX_POEORB_OPOD_20210812T111941_V20210722T225942_20210724T005942.EOF: 100%|█| 4.41M/4.41M [00:01<00:00,
Downloading products: 100%|██████████████████████████████████████████████████████████████| 1/1 [00:02<00:00,  2.42s/product]
Wrote new cfg file: GUNW_20210723-20210711_015000.yaml
WARNING: Weather model only extends to the surface topography; height levels below the topography will be interpolated from the surface and may be inaccurate.
Invalid extension GTiff for cube. Defaulting to .nc
Invalid extension GTiff for cube. Defaulting to .nc
Output cube spacing: 0.05 degrees
Output SNWE: [33.45, 35.45, -119.15, -115.95]
Starting to run the weather model calculation
Date: 20210723
Beginning weather model pre-processing
Weather model HRRR is available from 2016-07-15 to Present
Weather model HRRR is available from 2016-07-15 to Present
✅ Found ┊ model=hrrr ┊ product=nat ┊ 2021-Jul-23 01:00 UTC F00 ┊ GRIB2 @ aws ┊ IDX @ aws
/Users/cmarshak/mambaforge/envs/RAiDER/lib/python3.10/site-packages/herbie/core.py:797: UserWarning: This pattern is interpreted as a regular expression, and has match groups. To actually get the groups, use str.extract.
  logic = df.search_this.str.contains(searchString)
👨🏻‍🏭 Created directory: [/Users/cmarshak/Desktop/issue-ticket-gunw/weather_files/hrrr/20210723]
Note: Returning a list of [13] xarray.Datasets because cfgrib opened with multiple hypercubes.
Number of weather model nodes: 929214
Shape of weather model: (114, 143, 57)
Bounds of the weather model: -414306.15/-75306.15/-2070520.14/-1644520.14 (SNWE)
Weather model: HRRR
Mean value of the wet refractivity: 32.253670
Mean value of the hydrostatic refractivity: 168.645615

======Weather Model class object=====
Weather model time: 2021-07-23 01:00:00
Latitude resolution: 0.02702702702702703
Longitude resolution: 0.02702702702702703
Native projection: PROJCRS["unknown",BASEGEOGCRS["unknown",DATUM["unknown",ELLIPSOID["unknown",6371229,0,LENGTHUNIT["metre",1,ID["EPSG",9001]]]],PRIMEM["Greenwich",0,ANGLEUNIT["degree",0.0174532925199433],ID["EPSG",8901]]],CONVERSION["unknown",METHOD["Lambert Conic Conformal (2SP)",ID["EPSG",9802]],PARAMETER["Latitude of false origin",38.5,ANGLEUNIT["degree",0.0174532925199433],ID["EPSG",8821]],PARAMETER["Longitude of false origin",262.5,ANGLEUNIT["degree",0.0174532925199433],ID["EPSG",8822]],PARAMETER["Latitude of 1st standard parallel",38.5,ANGLEUNIT["degree",0.0174532925199433],ID["EPSG",8823]],PARAMETER["Latitude of 2nd standard parallel",38.5,ANGLEUNIT["degree",0.0174532925199433],ID["EPSG",8824]],PARAMETER["Easting at false origin",0,LENGTHUNIT["metre",1],ID["EPSG",8826]],PARAMETER["Northing at false origin",0,LENGTHUNIT["metre",1],ID["EPSG",8827]]],CS[Cartesian,2],AXIS["(E)",east,ORDER[1],LENGTHUNIT["metre",1,ID["EPSG",9001]]],AXIS["(N)",north,ORDER[2],LENGTHUNIT["metre",1,ID["EPSG",9001]]]]
ZMIN: -100.0
ZMAX: 80000.0
k1 = 0.776
k2 = 0.233
k3 = 3750.0
Humidity type = q
=====================================
Class name: hrrr
Dataset: hrrr
=====================================
A: []
B: []
Number of points in Lon/Lat = 114/143
Total number of grid points (3D): 929214
=====================================

Extent of the weather model is (xmin, ymin, xmax, ymax):-120.57, 32.53, -115.24, 36.33
Extent of the input is (xmin, ymin, xmax, ymax): -119.15, 33.45, -115.95, 35.45
Weather model HRRR is available from 2016-07-15 to Present
Weather model HRRR is available from 2016-07-15 to Present
✅ Found ┊ model=hrrr ┊ product=nat ┊ 2021-Jul-23 02:00 UTC F00 ┊ GRIB2 @ aws ┊ IDX @ aws
/Users/cmarshak/mambaforge/envs/RAiDER/lib/python3.10/site-packages/herbie/core.py:797: UserWarning: This pattern is interpreted as a regular expression, and has match groups. To actually get the groups, use str.extract.
  logic = df.search_this.str.contains(searchString)
Note: Returning a list of [13] xarray.Datasets because cfgrib opened with multiple hypercubes.
Number of weather model nodes: 929214
Shape of weather model: (114, 143, 57)
Bounds of the weather model: -414306.15/-75306.15/-2070520.14/-1644520.14 (SNWE)
Weather model: HRRR
Mean value of the wet refractivity: 32.389271
Mean value of the hydrostatic refractivity: 168.905441

======Weather Model class object=====
Weather model time: 2021-07-23 02:00:00
Latitude resolution: 0.02702702702702703
Longitude resolution: 0.02702702702702703
Native projection: PROJCRS["unknown",BASEGEOGCRS["unknown",DATUM["unknown",ELLIPSOID["unknown",6371229,0,LENGTHUNIT["metre",1,ID["EPSG",9001]]]],PRIMEM["Greenwich",0,ANGLEUNIT["degree",0.0174532925199433],ID["EPSG",8901]]],CONVERSION["unknown",METHOD["Lambert Conic Conformal (2SP)",ID["EPSG",9802]],PARAMETER["Latitude of false origin",38.5,ANGLEUNIT["degree",0.0174532925199433],ID["EPSG",8821]],PARAMETER["Longitude of false origin",262.5,ANGLEUNIT["degree",0.0174532925199433],ID["EPSG",8822]],PARAMETER["Latitude of 1st standard parallel",38.5,ANGLEUNIT["degree",0.0174532925199433],ID["EPSG",8823]],PARAMETER["Latitude of 2nd standard parallel",38.5,ANGLEUNIT["degree",0.0174532925199433],ID["EPSG",8824]],PARAMETER["Easting at false origin",0,LENGTHUNIT["metre",1],ID["EPSG",8826]],PARAMETER["Northing at false origin",0,LENGTHUNIT["metre",1],ID["EPSG",8827]]],CS[Cartesian,2],AXIS["(E)",east,ORDER[1],LENGTHUNIT["metre",1,ID["EPSG",9001]]],AXIS["(N)",north,ORDER[2],LENGTHUNIT["metre",1,ID["EPSG",9001]]]]
ZMIN: -100.0
ZMAX: 80000.0
k1 = 0.776
k2 = 0.233
k3 = 3750.0
Humidity type = q
=====================================
Class name: hrrr
Dataset: hrrr
=====================================
A: []
B: []
Number of points in Lon/Lat = 114/143
Total number of grid points (3D): 929214
=====================================

Extent of the weather model is (xmin, ymin, xmax, ymax):-120.57, 32.53, -115.24, 36.33
Extent of the input is (xmin, ymin, xmax, ymax): -119.15, 33.45, -115.95, 35.45
/Users/cmarshak/mambaforge/envs/RAiDER/lib/python3.10/site-packages/rasterio/__init__.py:304: NotGeoreferencedWarning: Dataset has no geotransform, gcps, or rpcs. The identity matrix will be returned.
  dataset = DatasetReader(path, driver=driver, sharing=sharing, **kwargs)
Processing slice 1 / 20: -500.0
Processing slice 2 / 20: 0.0
Processing slice 3 / 20: 500.0
Processing slice 4 / 20: 1000.0
Processing slice 5 / 20: 1500.0
Processing slice 6 / 20: 2000.0
Processing slice 7 / 20: 2500.0
Processing slice 8 / 20: 3000.0
Processing slice 9 / 20: 3500.0
Processing slice 10 / 20: 4000.0
Processing slice 11 / 20: 4500.0
Processing slice 12 / 20: 5000.0
Processing slice 13 / 20: 5500.0
Processing slice 14 / 20: 6000.0
Processing slice 15 / 20: 6500.0
Processing slice 16 / 20: 7000.0
Processing slice 17 / 20: 7500.0
Processing slice 18 / 20: 8000.0
Processing slice 19 / 20: 8500.0
Processing slice 20 / 20: 9000.0
CRITICAL: There are missing delay values. Check your inputs.

Successfully wrote delay cube to: /Users/cmarshak/Desktop/issue-ticket-gunw/HRRR_tropo_20210723T015024_ray.nc

Starting to run the weather model calculation
Date: 20210711
Beginning weather model pre-processing
Weather model HRRR is available from 2016-07-15 to Present
Weather model HRRR is available from 2016-07-15 to Present
✅ Found ┊ model=hrrr ┊ product=nat ┊ 2021-Jul-11 01:00 UTC F00 ┊ GRIB2 @ aws ┊ IDX @ aws
/Users/cmarshak/mambaforge/envs/RAiDER/lib/python3.10/site-packages/herbie/core.py:797: UserWarning: This pattern is interpreted as a regular expression, and has match groups. To actually get the groups, use str.extract.
  logic = df.search_this.str.contains(searchString)
👨🏻‍🏭 Created directory: [/Users/cmarshak/Desktop/issue-ticket-gunw/weather_files/hrrr/20210711]
Note: Returning a list of [13] xarray.Datasets because cfgrib opened with multiple hypercubes.
Number of weather model nodes: 929214
Shape of weather model: (114, 143, 57)
Bounds of the weather model: -414306.15/-75306.15/-2070520.14/-1644520.14 (SNWE)
Weather model: HRRR
Mean value of the wet refractivity: 29.419880
Mean value of the hydrostatic refractivity: 166.646515

======Weather Model class object=====
Weather model time: 2021-07-11 01:00:00
Latitude resolution: 0.02702702702702703
Longitude resolution: 0.02702702702702703
Native projection: PROJCRS["unknown",BASEGEOGCRS["unknown",DATUM["unknown",ELLIPSOID["unknown",6371229,0,LENGTHUNIT["metre",1,ID["EPSG",9001]]]],PRIMEM["Greenwich",0,ANGLEUNIT["degree",0.0174532925199433],ID["EPSG",8901]]],CONVERSION["unknown",METHOD["Lambert Conic Conformal (2SP)",ID["EPSG",9802]],PARAMETER["Latitude of false origin",38.5,ANGLEUNIT["degree",0.0174532925199433],ID["EPSG",8821]],PARAMETER["Longitude of false origin",262.5,ANGLEUNIT["degree",0.0174532925199433],ID["EPSG",8822]],PARAMETER["Latitude of 1st standard parallel",38.5,ANGLEUNIT["degree",0.0174532925199433],ID["EPSG",8823]],PARAMETER["Latitude of 2nd standard parallel",38.5,ANGLEUNIT["degree",0.0174532925199433],ID["EPSG",8824]],PARAMETER["Easting at false origin",0,LENGTHUNIT["metre",1],ID["EPSG",8826]],PARAMETER["Northing at false origin",0,LENGTHUNIT["metre",1],ID["EPSG",8827]]],CS[Cartesian,2],AXIS["(E)",east,ORDER[1],LENGTHUNIT["metre",1,ID["EPSG",9001]]],AXIS["(N)",north,ORDER[2],LENGTHUNIT["metre",1,ID["EPSG",9001]]]]
ZMIN: -100.0
ZMAX: 80000.0
k1 = 0.776
k2 = 0.233
k3 = 3750.0
Humidity type = q
=====================================
Class name: hrrr
Dataset: hrrr
=====================================
A: []
B: []
Number of points in Lon/Lat = 114/143
Total number of grid points (3D): 929214
=====================================

Extent of the weather model is (xmin, ymin, xmax, ymax):-120.57, 32.53, -115.24, 36.33
Extent of the input is (xmin, ymin, xmax, ymax): -119.15, 33.45, -115.95, 35.45
Weather model HRRR is available from 2016-07-15 to Present
Weather model HRRR is available from 2016-07-15 to Present
✅ Found ┊ model=hrrr ┊ product=nat ┊ 2021-Jul-11 02:00 UTC F00 ┊ GRIB2 @ aws ┊ IDX @ aws
/Users/cmarshak/mambaforge/envs/RAiDER/lib/python3.10/site-packages/herbie/core.py:797: UserWarning: This pattern is interpreted as a regular expression, and has match groups. To actually get the groups, use str.extract.
  logic = df.search_this.str.contains(searchString)
Note: Returning a list of [13] xarray.Datasets because cfgrib opened with multiple hypercubes.
Number of weather model nodes: 929214
Shape of weather model: (114, 143, 57)
Bounds of the weather model: -414306.15/-75306.15/-2070520.14/-1644520.14 (SNWE)
Weather model: HRRR
Mean value of the wet refractivity: 28.742548
Mean value of the hydrostatic refractivity: 166.787643

======Weather Model class object=====
Weather model time: 2021-07-11 02:00:00
Latitude resolution: 0.02702702702702703
Longitude resolution: 0.02702702702702703
Native projection: PROJCRS["unknown",BASEGEOGCRS["unknown",DATUM["unknown",ELLIPSOID["unknown",6371229,0,LENGTHUNIT["metre",1,ID["EPSG",9001]]]],PRIMEM["Greenwich",0,ANGLEUNIT["degree",0.0174532925199433],ID["EPSG",8901]]],CONVERSION["unknown",METHOD["Lambert Conic Conformal (2SP)",ID["EPSG",9802]],PARAMETER["Latitude of false origin",38.5,ANGLEUNIT["degree",0.0174532925199433],ID["EPSG",8821]],PARAMETER["Longitude of false origin",262.5,ANGLEUNIT["degree",0.0174532925199433],ID["EPSG",8822]],PARAMETER["Latitude of 1st standard parallel",38.5,ANGLEUNIT["degree",0.0174532925199433],ID["EPSG",8823]],PARAMETER["Latitude of 2nd standard parallel",38.5,ANGLEUNIT["degree",0.0174532925199433],ID["EPSG",8824]],PARAMETER["Easting at false origin",0,LENGTHUNIT["metre",1],ID["EPSG",8826]],PARAMETER["Northing at false origin",0,LENGTHUNIT["metre",1],ID["EPSG",8827]]],CS[Cartesian,2],AXIS["(E)",east,ORDER[1],LENGTHUNIT["metre",1,ID["EPSG",9001]]],AXIS["(N)",north,ORDER[2],LENGTHUNIT["metre",1,ID["EPSG",9001]]]]
ZMIN: -100.0
ZMAX: 80000.0
k1 = 0.776
k2 = 0.233
k3 = 3750.0
Humidity type = q
=====================================
Class name: hrrr
Dataset: hrrr
=====================================
A: []
B: []
Number of points in Lon/Lat = 114/143
Total number of grid points (3D): 929214
=====================================

Extent of the weather model is (xmin, ymin, xmax, ymax):-120.57, 32.53, -115.24, 36.33
Extent of the input is (xmin, ymin, xmax, ymax): -119.15, 33.45, -115.95, 35.45
/Users/cmarshak/mambaforge/envs/RAiDER/lib/python3.10/site-packages/rasterio/__init__.py:304: NotGeoreferencedWarning: Dataset has no geotransform, gcps, or rpcs. The identity matrix will be returned.
  dataset = DatasetReader(path, driver=driver, sharing=sharing, **kwargs)
Processing slice 1 / 20: -500.0
Processing slice 2 / 20: 0.0
Processing slice 3 / 20: 500.0
Processing slice 4 / 20: 1000.0
Processing slice 5 / 20: 1500.0
Processing slice 6 / 20: 2000.0
Processing slice 7 / 20: 2500.0
Processing slice 8 / 20: 3000.0
Processing slice 9 / 20: 3500.0
Processing slice 10 / 20: 4000.0
Processing slice 11 / 20: 4500.0
Processing slice 12 / 20: 5000.0
Processing slice 13 / 20: 5500.0
Processing slice 14 / 20: 6000.0
Processing slice 15 / 20: 6500.0
Processing slice 16 / 20: 7000.0
Processing slice 17 / 20: 7500.0
Processing slice 18 / 20: 8000.0
Processing slice 19 / 20: 8500.0
Processing slice 20 / 20: 9000.0
CRITICAL: There are missing delay values. Check your inputs.

Successfully wrote delay cube to: /Users/cmarshak/Desktop/issue-ticket-gunw/HRRR_tropo_20210711T015024_ray.nc

Wrote slc delays to: /Users/cmarshak/Desktop/issue-ticket-gunw/HRRR_interferometric_20210723-20210711_015000.nc
Updated troposphere group in: S1-GUNW-A-R-064-tops-20210723_20210711-015000-00119W_00033N-PP-6267-v2_0_6.nc
```

### PR DESCRIPTION
**Pull request by bbuzz31**
Nan values are introduced when we attempt to integrate rays above the weather model ceiling. 
This change will force the top of the integration to be 1 meter less than the top of the weather model.
Closes #574

### CODE REVIEW

**Comment by dbekaert** [comment]
@bbuzz31 there needs to be more transparancy for the user, if they had set max height higher and you are overwriting. Im wondering if we should also make this a metadata variable that we cary. It would explain model differences between models in the absolute delays.

**Comment by piyushrpt** [comment]
Hypothetical situation. If we have 3 frames along the track and the same weather model source file is used for generating the delays. If the max height is determined on a frame by frame basis, they will be different. Is there a benefit to considering a constant max height to ensure the delays dont have discontinuities?

**Comment by dbekaert** [comment]
I thought we hardcoded the model heights for the globe or conus based on
what the average height was for each model model level. That however will
sometimes be above and other times below the actual model height. Brett
could you link to that part of the code?

Perhaps a test to do would be to be conservative amd compute what the min
height level would be per model level?

So as long as the above holds aka hardcoded value per model over the model
geographical extend there should not be a stitch artifact i believe.

**Comment by bbuzz31** [comment]
@dbekaert the constant height level was 15km until recently (June) when we changed it to 80km to integrate (`_ZREF` in constants.py)

The height levels in `model_levels.py` are the geometric heights we interpolate the geopotential model heights to. The ~26km top level of HRRR is the lowest one and doesn't vary spatially over CONUS because we interpolate the native weather models to a regular z grid. Same for the globe with GMAO, so I don't think there should be stitching problems

File: tools/RAiDER/constants.py [hunk 1]
```diff
@@ -8,7 +8,7 @@
 import numpy as np
 
 _ZMIN = np.float64(-100)   # minimum required height
-_ZREF = np.float64(80000)  # maximum integration height
+_ZREF = np.float64(26000)  # maximum integration height
```
Comment (jlmaurer) [inline tools/RAiDER/constants.py:11]: should we say "when not specified by user"?

**Review by jlmaurer — COMMENTED** [review]
LGTM except for one minor comment.

**Review by jlmaurer — APPROVED** [review]

**Pull request merged**